x-pack/solutions/search/plugins/search_indices/public/components/quick_stats/quick_stats.tsx

```diff
@@ -8,168 +8,93 @@
-        <EuiFlexItem grow={false}>
-          <EuiText>
-            <h6>
-              {i18n.translate('xpack.searchIndices.quickStats.setup_ai_search_description', {
-                defaultMessage: 'Build AI-powered search experiences with Elastic',
-              })}
-            </h6>
-          </EuiText>
-        </EuiFlexItem>
-        <EuiFlexItem grow={false}>
-          <EuiButton
-            href={docLinks.links.enterpriseSearch.semanticSearch}
-            target="_blank"
-            data-test-subj="setupAISearchButton"
-          >
-            {i18n.translate('xpack.searchIndices.quickStats.setup_ai_search_button', {
-              defaultMessage: 'Set up now',
-            })}
-          </EuiButton>
-        </EuiFlexItem>
-      </EuiFlexGroup>
-    </EuiPanel>
-  );
-};
-
-export const QuickStats: React.FC<QuickStatsProps> = ({ index, mappings, indexDocuments }) => {
+export const QuickStats: React.FC<QuickStatsProps> = ({
+  index,
+  mappings,
+  indexDocuments,
+  isStateless,
+}) => {
   const [open, setOpen] = useState<boolean>(false);
   const { euiTheme } = useEuiTheme();
-  const mappingStats = useMemo(() => countVectorBasedTypesFromMappings(mappings), [mappings]);
-  const vectorFieldCount =
-    mappingStats.sparse_vector + mappingStats.dense_vector + mappingStats.semantic_text;
-  const docCount = indexDocuments?.results._meta.page.total ?? 0;
+  const { mappingStats, vectorFieldCount } = useMemo(() => {
+    const stats = countVectorBasedTypesFromMappings(mappings);
+    const vectorFields = stats.sparse_vector + stats.dense_vector + stats.semantic_text;
+    return { mappingStats: stats, vectorFieldCount: vectorFields };
+  }, [mappings]);
+
+  const stats = isStateless
+    ? [
+        <StatelessDocumentCountStat
+          index={index}
+          documentCount={indexDocuments?.results._meta.page.total ?? 0}
+          open={open}
+          setOpen={setOpen}
+        />,
+        ...(Array.isArray(index.aliases) && index.aliases.length > 0
+          ? [<AliasesStat aliases={index.aliases} open={open} setOpen={setOpen} />]
+          : []),
+        <AISearchQuickStat
+          mappingStats={mappingStats}
+          vectorFieldCount={vectorFieldCount}
+          open={open}
+          setOpen={setOpen}
+        />,
+      ]
+    : [
+        <IndexStatusStat index={index} open={open} setOpen={setOpen} />,
+        <StatefulIndexStorageStat index={index} open={open} setOpen={setOpen} />,
+        <StatefulDocumentCountStat
+          open={open}
+          setOpen={setOpen}
+          index={index}
+          mappingStats={mappingStats}
+        />,
+        ...(Array.isArray(index.aliases) && index.aliases.length > 0
+          ? [<AliasesStat aliases={index.aliases} open={open} setOpen={setOpen} />]
+          : []),
+        <AISearchQuickStat
+          mappingStats={mappingStats}
+          vectorFieldCount={vectorFieldCount}
+          open={open}
+          setOpen={setOpen}
+        />,
```

Comment:
nitpick: maybe let's move this code to `baseStats` value to avoid duplications?

Reply:
which code? the definition of which stats are displayed for stateless vs stateful?

Reply:
Yep, this two stats seems the same for both stateless and statefull

Reply:
They are for now, but I wanted it to be clear in this one spot which stats are used by stateless & stateful without having to look anywhere else. And if we end up need to replace them for environment specific versions in the future it would stay here as well.